File: src/modules/declarationGenerator.ts
/**
 * 声明文件生成器
 * 用于生成组件和自动导入的声明文件
 */

import fs from "fs-extra";
import path from "path";

/**
 * 生成声明文件
 * @param projectPath 项目路径
 */
export async function generateDeclarationFiles(
  projectPath: string
): Promise<void> {
  const declarationFiles = [
    {
      name: "components.d.ts",
      content: `// 自动生成的组件声明文件
// 由ew-auto-import-tool生成，请勿手动修改

declare module 'vue' {
  export interface GlobalComponents {
    // 将在项目构建时自动填充
  }
}

export {}
`,
    },
    {
      name: "auto-imports.d.ts",
      content: `// 自动生成的API导入声明文件
// 由ew-auto-import-tool生成，请勿手动修改

declare global {
  // 将在项目构建时自动填充
}

export {}
`,
    },
  ];

  for (const file of declarationFiles) {
    const filePath = path.join(projectPath, file.name);

    // 检查文件是否已存在
    if (fs.existsSync(filePath)) {
      console.log(`声明文件已存在，跳过生成: ${filePath}`);
      continue;
    }

    // 写入声明文件
    await fs.writeFile(filePath, file.content, "utf-8");
    console.log(`已生成声明文件: ${filePath}`);
  }
}

File: rust/src/utils/declaration_generator.rs
use anyhow::{Context, Result};
use std::fs;
use std::path::Path;

/// 生成声明文件
pub fn generate_declaration_files(project_path: &Path) -> Result<()> {
    let declaration_files = [
        (
            "components.d.ts",
            "// 自动生成的组件声明文件
// 由ew-auto-import-tool生成，请勿手动修改

declare module 'vue' {
  export interface GlobalComponents {
    // 将在项目构建时自动填充
  }
}

export {}
",
        ),
        (
            "auto-imports.d.ts",
            "// 自动生成的API导入声明文件
// 由ew-auto-import-tool生成，请勿手动修改

declare global {
  // 将在项目构建时自动填充
}

export {}
",
        ),
    ];

    for (file_name, content) in declaration_files.iter() {
        let file_path = project_path.join(file_name);

        // 检查文件是否已存在
        if file_path.exists() {
            println!("声明文件已存在，跳过生成: {:?}", file_path);
            continue;
        }

        // 写入文件
        fs::write(&file_path, content)
            .context(format!("创建声明文件失败: {:?}", file_path))?;

        println!("已生成声明文件: {:?}", file_path);
    }

    Ok(())
}